### PR TITLE
Added /producer/by_consumer metrics.

### PR DESCRIPTION
**Pull request by mangchiandjjoe**
Including "set consumer_id to producer project id for Quota call"

### CODE REVIEW

File: contrib/endpoints/src/api_manager/service_control/proto.cc [hunk 1]
```diff
@@ -270,6 +278,13 @@ const SupportedMetric supported_metrics[] = {
         set_distribution_metric_to_request_size,
     },
     {
+        "serviceruntime.googleapis.com/api/producer/by_consumer/request_sizes",
+        ::google::api::MetricDescriptor_MetricKind_DELTA,
+        ::google::api::MetricDescriptor_ValueType_DISTRIBUTION,
+        SupportedMetric::FINAL, SupportedMetric::PRODUCER,
+        set_distribution_metric_to_response_size,
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/service_control/proto.cc]: to_request_size?

File: contrib/endpoints/src/api_manager/service_control/proto.cc [hunk 2]
```diff
@@ -959,6 +1012,9 @@ utils::Status Proto::FillAllocateQuotaRequest(
   if (!info.api_key.empty()) {
     operation->set_consumer_id(std::string(kConsumerIdApiKey) +
                                std::string(info.api_key));
+  } else if (!info.producer_project_id.empty()) {
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/service_control/proto.cc]: is this change part of this PR?

File: contrib/endpoints/src/api_manager/service_control/proto_test.cc [hunk 3]
```diff
@@ -229,6 +229,43 @@ TEST_F(ProtoTest, FillAllocateQuotaRequestNoMethodNameTest) {
   ASSERT_EQ(expected_text, text);
 }
 
+TEST_F(ProtoTest, FillNoApiKeyAllocateQuotaRequestTest) {
+  std::vector<std::pair<std::string, int>> metric_cost_vector = {
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/service_control/proto_test.cc]: this should not be part of this PR

File: contrib/endpoints/src/api_manager/service_control/proto.cc (hunk 2 above)
Comment (mangchiandjjoe) [inline contrib/endpoints/src/api_manager/service_control/proto.cc]: I will separated the PR

File: contrib/endpoints/src/api_manager/service_control/proto_test.cc (hunk 3 above)
Comment (mangchiandjjoe) [inline contrib/endpoints/src/api_manager/service_control/proto_test.cc]: I will separated the PR

**Comment by mangchiandjjoe** [comment]
Closing  the PR to separate changes
